### PR TITLE
Log metrics for GPUs in use by the training process separately

### PR DESCRIPTION
**Pull request by davidwallacejackson**
If a GPU is being used by the current process (or its parent, accounting for the forking off of our logging utility), all of its metrics will be logged a second time as `system/gpu.process.{index}.{metric}`.

You wind up with a whole bunch of metrics:

image

It would be better to log these metrics in a non-duplicative way, but this at least makes the data available.

### CODE REVIEW

**Comment by vanpelt** [comment]
I’m not quite following, are we now duplicating metrics if a GPU is in use?

**Comment by vanpelt** [comment]
I would think the ideal solution would be to store the indices of the in use GPU’s in something like wandb.config._wandb and just make non in use GPU’s hidden by default.

**Comment by vanpelt** [comment]
@shawnlewis thoughts?

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+    "python.testing.pytestEnabled": true,
+    "python.formatting.provider": "autopep8",
+    "python.formatting.autopep8Args": [
+        "--ignore-local-config",
+        "--ignore",
+        "E265,E402,E202,E225,E226",
+        "--max-line-length",
+        "120"
+    ],
+    "python.pythonPath": "/Users/davidjackson/.pyenv/versions/3.6.8/bin/python"
```
Comment (raubitsj) [inline .vscode/settings.json]: revert this file please.

File: wandb/stats.py [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+import os
 from numbers import Number
 import threading
 from wandb import util
 from wandb import termlog
 psutil = util.get_module("psutil")
 
+def gpu_in_use_by_this_process(gpu_handle):
+    our_pids = os.getpid()
+    our_parent_pid = os.getppid()
```
Comment (raubitsj) [inline wandb/stats.py]: Not supported in windows on python 2.7

And in general I am not sure about this approach, what happens if the user is using data parallel pytorch and spinning up multiple processes?

File: .vscode/settings.json (hunk 1 above)
Comment (davidwallacejackson) [reply .vscode/settings.json]: done -- sorry about that.

The change I intended to make (and still think we should, if I can figure out how to do it) is to allow local-env overriding of the python path. This settings file is super high-priority, so it takes precedence over everything else.

**Comment by raubitsj** [comment]
> I would think the ideal solution would be to store the indices of the in use GPU’s in something like wandb.config._wandb and just make non in use GPU’s hidden by default.

I think the benefit is that this approach handles the resume case where a run moves between machines.  The consequence is more logging.

File: wandb/stats.py (hunk 2 above)
Comment (davidwallacejackson) [reply wandb/stats.py]: Updated to use psutil, and to recursively list all child pids. Still possible edge cases around `wandb run`, as we discussed offline, but much more robust overall.

**Pull request merged**